include/ck_tile/ops/gemm/kernel/gemm_kernel.hpp

```diff
@@ -276,68 +275,77 @@ struct GemmKernel
-        const auto c_block_window = make_tile_window(
+        auto&& c_pad_view     = views.at(I2);
+        auto&& c_block_window = make_tile_window(
             c_pad_view,
             make_tuple(number<TilePartitioner::kM>{}, number<TilePartitioner::kN>{}),
             {i_m, i_n});
 
         return make_tuple(a_block_window, b_block_window, c_block_window);
     }
 
-    CK_TILE_DEVICE void run_common_gemm_pipeline(const ADataType* a_start,
-                                                 const BDataType* b_start,
-                                                 CDataType* c_start,
-                                                 const GemmCommonKargs& kargs,
-                                                 const index_t i_m,
-                                                 const index_t i_n) const
+    /**
+     * Create tensor views, pad views, tile windows.
+     * Runs GEMM cooperatively by whole workgroup with CShuffle or Default 2D Epilogue
+     *
+     * @param a_ptr input A pointer
+     * @param b_ptr input B pointer
+     * @param c_ptr output C pointer
+     * @param kargs GEMM kernel arguments
+     * @param block_idx_m M block index
+     * @param block_idx_n N block index
+     */
+    CK_TILE_DEVICE void RunGemm(const ADataType* a_ptr,
+                                const BDataType* b_ptr,
+                                CDataType* c_ptr,
+                                const GemmKernelArgs& kargs,
+                                const index_t block_idx_m,
+                                const index_t block_idx_n) const
     {
-        // Convert pointers to tensor views
-        const auto gemm_tensor_views_tuple =
-            make_gemm_tensor_views(a_start, b_start, c_start, kargs);
-        const auto gemm_pad_views    = make_gemm_pad_views(gemm_tensor_views_tuple);
-        const auto gemm_tile_windows = make_gemm_tile_windows(gemm_pad_views, i_m, i_n);
+        // Create Gemm tensor views, pad views and tile windows
+        auto&& gemm_tensor_views_tuple = MakeGemmTensorViews(a_ptr, b_ptr, c_ptr, kargs);
+        auto&& gemm_pad_views          = MakeGemmPadViews(gemm_tensor_views_tuple);
+        auto&& gemm_tile_windows = MakeGemmTileWindows(gemm_pad_views, block_idx_m, block_idx_n);
 
         // allocate LDS
         __shared__ char smem_ptr[GetSmemSize()];
 
         const index_t num_loop = TilePartitioner::GetLoopNum(kargs.K);
 
-        auto a_block_window = gemm_tile_windows.at(I0);
-        auto b_block_window = gemm_tile_windows.at(I1);
-
         // Run GEMM cooperatively by whole workgroup.
-        auto c_block_tile =
+        auto&& a_block_window = gemm_tile_windows.at(I0);
+        auto&& b_block_window = gemm_tile_windows.at(I1);
+        auto&& c_block_tile =
             GemmPipeline{}.template operator()(a_block_window, b_block_window, num_loop, smem_ptr);
 
-        auto c_block_window = gemm_tile_windows.at(I2);
-
+        // Run CShuffle or Default 2D Epilogue
```

Comment:
There maybe plethora of possible epilogues. 
```suggestion
        // Run Epilogue pipeline. 
```